tests/backup/backup_restore_basic_test.go

```diff
@@ -954,10 +954,13 @@ var _ = Describe("{AllNSBackupWithIncludeNewNSOption}", func() {
 		allAppNamespaces := append(appNamespaces, newAppNamespaces...)
 		log.InfoD("Deleting deployed namespaces - %v", allAppNamespaces)
 		ValidateAndDestroy(allContexts, opts)
-
 		err := SetSourceKubeConfig()
 		log.FailOnError(err, "failed to switch context to source cluster")
-
+		// TO DO: Remove suspendBackupSchedule after PB-5221 is fixed
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		err = suspendBackupSchedule(scheduleName, schedulePolicyName, orgID, ctx)
```

Comment:
Can we add suspend inside the delete schedule function itself. So that we are making sure the schedule is suspended before deletion? Later we can remove it from that function.
Now we don't know in which other test cases we are deleting the schedule without suspending it.

Reply:
I also thought so , but I tried doing it manually, if its already in suspended state, there will be no option  of suspend again , only resume option
So will api also not throw error if the testcase has already suspended it?
Also its only this testcase in last many runs that has  been failing because of this issue

Reply:
Then we can add a check before we suspend it. If it is suspended then we will directly delete. If not, we will suspend and delete.
If someone else uses delete in future they should not hit this issue. That's the intention here.

If you feel we should not be changing the existing delete schedule function, then you can create a new function called SuspendAndDeleteSchedule and use that in the test case. It will be a useful function.

Reply:
Done